src/api/informer.rs

```diff
@@ -203,13 +211,15 @@ where
     pub async fn reset(&self) -> Result<()> {
         // Fetch a new initial version:
         let initial = self.get_resource_version().await?;
-        *self.version.write().unwrap() = initial;
+        *self.version.lock().await = initial;
         Ok(())
     }
 
     /// Return the current version
     pub fn version(&self) -> String {
-        self.version.read().unwrap().clone()
+        // We need to block on a future here quickly
+        // to get a lock on our version
+        futures::executor::block_on(async { self.version.lock().await.clone() })
```

Comment:
This could arguably be an async fn as it could block if people use the informer from non-linear contexts, and in that case it might be better to expose the future. For now, I can't imagine such a scenario though.